### ISSUE TITLE
startkey/endkey view generation

### ISSUE THREAD

**Issue by robertkeizer** [pouchdb/pouchdb#2928]
I'm dealing with a stupidly large number of documents in Pouch; On the order of ~20k or more.

Views are painful, as every document given a since is mapped in, or at the very least, hits a filter function.

I've setup my documents and views in such a way that the ability to specify a startkey and endkey for what documents are used in view generation would be handy.

This would require a change from .changes in `updateView` to allDocs. #2927 is related in this respect.

pouchdb-mapreduce may be a more appropriate place for this, as that is where the change will have to be made, although I feel as though any discussion should take place in this project.

Thoughts? Would a pull request for this functionality be accepted? I'm not sure exactly how to expose these options out, I'm rolling my own little solution, but would like to contribute back if this would help others as well.

tl;dr: By switching to `allDocs` with a startkey/endkey for view generation, instead of .changes and a filter function, views that exist in a database with a very high number of documents, that aren't related to the view can be omitted, brining the view generation time down to something reasonable.

**Comment by daleharvey** [comment]
Given our general shift is away from map / reduce, closing out any older feature requests on it for now
